### PR TITLE
*: `query-status -s` and purge-relay send to all relay worker

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
`query-status -s` and purge-relay send to all relay worker

### Check List 

Tests 

 - Unit test
 - Integration test

Code changes

Side effects

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation
 - Need to be included in the release note

### CODE REVIEW

**Comment by lance6716** [comment]
test failed because history bound is prior to relay worker, will fix soon

File: dm/master/server.go [hunk 1]
```diff
@@ -895,43 +925,79 @@ func (s *Server) getStatusFromWorkers(ctx context.Context, sources []string, tas
...
+			}
+		}
+
+		if len(workers) == 0 {
+			err := terror.ErrMasterWorkerArgsExtractor.Generatef("%s relevant worker-client not found", source)
+			handleErr(err, source, "")
+			continue
+		}
+
+		for _, worker := range workers {
```
Comment (lichunzhu) [inline dm/master/server.go:967]: Will we get repeat status from worker here?

File: tests/new_relay/run.sh [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+    # after restarting, worker will purge relay log directory because checkpoint is newer than relay.meta
+    run_dm_worker $WORK_DIR/worker1 $WORKER1_PORT $cur/conf/dm-worker1.toml
+    check_rpc_alive $cur/../bin/check_worker_online 127.0.0.1:$WORKER1_PORT
+
+    run_sql_file $cur/data/db1.increment3.sql $MYSQL_HOST1 $MYSQL_PORT1 $MYSQL_PASSWORD1
+    check_sync_diff $WORK_DIR $cur/conf/diff_config.toml
+
+    run_dm_ctl_with_retry $WORK_DIR "127.0.0.1:$MASTER_PORT" \
+        "query-status -s $SOURCE_ID1" \
```
Comment (lichunzhu) [inline tests/new_relay/run.sh:67]: Could we check more about the result of `query-status`?

**Review by lichunzhu — COMMENTED** [review]
Rest LGTM.
Besides, CI failed again

File: dm/master/server.go (hunk 1 above)
Comment (lance6716) [reply dm/master/server.go:967]: we use `workerNameSet` to avoid repeating

File: tests/new_relay/run.sh (hunk 2 above)
Comment (lance6716) [reply tests/new_relay/run.sh:67]: what's your expected checking 🤔 Now we checked the two workers has started relay (has true result)

**Comment by lance6716** [comment]
CI failed:

I don't know why there's only one "Welcome to dm-worker" in worker4.log, but

[2021-04-08T03:22:51.586Z] [2021/04/08 11:22:29.931 +08:00] [INFO] [main.go:84] ["got signal to exit"] [signal=terminated]
[2021-04-08T03:22:51.586Z] [2021/04/08 11:22:30.354 +08:00] [INFO] [main.go:92] ["dm-worker exit"]
[2021-04-08T03:22:51.586Z] [2021/04/08 11:22:30.380 +08:00] [INFO] [main.go:84] ["got signal to exit"] [signal=terminated]

File: tests/new_relay/run.sh (hunk 2 above)
Comment (lichunzhu) [reply tests/new_relay/run.sh:67]: OK. LGTM

File: dm/master/server.go (hunk 1 above)
Comment (lichunzhu) [reply dm/master/server.go:967]: Since every worker only bound to one source, It LGTM now.

**Comment by lance6716** [comment]
seems this time CI will success https://internal.pingcap.net/idc-jenkins/blue/organizations/jenkins/dm_ghpr_test/detail/dm_ghpr_test/8760/pipeline/76

I'll re-run many times to check if it's fixed.

**Comment by lance6716** [comment]
@GMHDBJD PTAL

**Review by GMHDBJD — APPROVED** [review]
/lgtm

**Pull request merged**